packages/valory/skills/agent_performance_summary_abci/behaviours.py

```diff
@@ -154,8 +158,33 @@ def calculate_roi(self):
 
     def _get_prediction_accuracy(self):
         """Get the prediction accuracy."""
-        # TODO: implement real accuracy fetching
-        return None
+
+        agent_id = self.context.agent_address.lower()
+        agent_bets_data = yield from self._fetch_trader_agent_bets(
+            agent_id=agent_id,
+        )
+        if agent_bets_data is None or len(agent_bets_data["bets"]) == 0:
+            return None
+
+        bets_on_closed_markets = [
+            bet
+            for bet in agent_bets_data["bets"]
+            if bet["fixedProductMarketMaker"]["currentAnswer"] is not None
+        ]
+        total_bets = len(bets_on_closed_markets)
+        won_bets = 0
+
+        for bet in bets_on_closed_markets:
+            market_answer = bet["fixedProductMarketMaker"]["currentAnswer"]
+            bet_answer = bet["outcomeIndex"]
+            if market_answer == INVALID_ANSWER_HEX:
+                continue
+            if int(market_answer, 0) == int(bet_answer):
```

Comment:
```suggestion
            if int(market_answer, 0) == int(bet_answer, 16):
```

Otherwise you'd get 

> int("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'

Reply:
Is the `0xffff...` a special case. So far, I've only seen bet_answer as '0' or '1', and with your suggested change the code still works correctly for those. I am trying to confirm if this fix is mainly to catch edge cases like these. Sharing some logs for context

```
[2025-09-22 18:23:18,638] [INFO] [trader] market_answer='0x0000000000000000000000000000000000000000000000000000000000000000' bet_answer='0' int(bet_answer,16)=0
[2025-09-22 18:23:18,638][INFO] [trader] market_answer='0x0000000000000000000000000000000000000000000000000000000000000000' bet_answer='0' int(bet_answer,16)=0
[2025-09-22 18:23:18,638] [INFO] [trader] market_answer='0x0000000000000000000000000000000000000000000000000000000000000001' bet_answer='1' int(bet_answer,16)=1
[2025-09-22 18:23:18,638][INFO] [trader] market_answer='0x0000000000000000000000000000000000000000000000000000000000000001' bet_answer='1' int(bet_answer,16)=1
[2025-09-22 18:23:18,638] [INFO] [trader] market_answer='0x0000000000000000000000000000000000000000000000000000000000000000' bet_answer='0' int(bet_answer,16)=0
```

Reply:
It's about type of str to int. With any other hex string it would be the same:

```
int("0x0000000000000000000000000000000000000000000000000000000000000000")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '0x0000000000000000000000000000000000000000000000000000000000000000'
```

Reply:
My bad, it's covered by `int(market_answer, 0)` not `int(bet_answer)`